widget/button.go

```diff
@@ -369,37 +378,38 @@ func (r *buttonRenderer) applyTheme() {
 	}
 	r.label.Refresh()
 	if r.icon != nil && r.icon.Resource != nil {
+		icon := r.icon.Resource
 		switch res := r.icon.Resource.(type) {
-		case *theme.ThemedResource:
-			if r.button.Importance == HighImportance || r.button.Importance == DangerImportance || r.button.Importance == WarningImportance || r.button.Importance == SuccessImportance {
-				r.icon.Resource = theme.NewInvertedThemedResource(res)
-				r.icon.Refresh()
-			}
 		case *theme.InvertedThemedResource:
 			if r.button.Importance != HighImportance && r.button.Importance != DangerImportance && r.button.Importance != WarningImportance && r.button.Importance != SuccessImportance {
-				r.icon.Resource = res.Original()
-				r.icon.Refresh()
+				icon = res.Original()
+			}
+		case *theme.ThemedResource, fyne.ThemedResource:
+			if r.button.Importance == HighImportance || r.button.Importance == DangerImportance || r.button.Importance == WarningImportance || r.button.Importance == SuccessImportance {
+				icon = theme.NewInvertedThemedResource(res)
 			}
 		}
+		r.icon.Resource = cache.OverrideResourceTheme(icon, r.button)
```

Comment:
I'm still a bit worried about the fact ghat a custom widget wouldn't be able to set the resource this way given that the `cache` package is internal.

Reply:
I can expose it through a public method on `container.ThemeOverride` like `AdoptResource` or `OverrideResource`?

Reply:
That sounds like a good plan 👍

Reply:
Hmm. actually it can't be there - that is handy for app developers not custom widget developers.
`BaseWidget` is a place it could be - but then every single widget would have that function on it.
The `theme` package remains an option but I think that goes against moving to the container type recently.

I am thinking that this maybe benefits from being part of the "could be done from the render stack" which came up on a call but is an additional layer of work/refactoring. That way any resource in an overridden container could be interpreted that way in a future PR, removing the need for any more public API?

Reply:
Good point

Reply:
That does sound like an even better solution although I understand that it will require quite a lot of work